File: benchmarks/primegeneration/PrimeGeneration.java
public class PrimeGeneration {
    // Configuration - limits for counting
    private static final int LIMIT1 = 100;
    private static final int LIMIT2 = 1000;
    private static final int LIMIT3 = 10000;
    private static final int LIMIT4 = 50000;

    // Markers matching Lamina version
    private static final long HEADER_MARKER = 123456789L;
    private static final long FOOTER_MARKER = 987654321L;

    public static boolean isPrime(int n) {
        if (n <= 1) return false;
        if (n <= 3) return true;
        if (n % 2 == 0) return false;

        for (int i = 3; i * i <= n; i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }

    public static long countPrimes(int limit) {
        if (limit < 2) return 0;
        long count = 0;
        for (int i = 2; i <= limit; i++) {
            if (isPrime(i)) count++;
        }
        return count;
    }

    public static void main(String[] args) {
        // Print header marker
        System.out.println(HEADER_MARKER);

        // Count and print prime counts for different limits
        System.out.println(countPrimes(LIMIT1));
        System.out.println(countPrimes(LIMIT2));
        System.out.println(countPrimes(LIMIT3));
        System.out.println(countPrimes(LIMIT4));

        // Print footer marker
        System.out.println(FOOTER_MARKER);
    }
}

File: benchmarks/primegeneration/primegeneration.py
# primegeneration.py
# Python implementation of prime generation benchmark (simplified counting)

# Configuration - limits for counting
LIMIT1 = 100
LIMIT2 = 1000
LIMIT3 = 10000
LIMIT4 = 50000

# Markers matching Lamina version
HEADER_MARKER = 123456789
FOOTER_MARKER = 987654321

def is_prime(n):
    """Check if a number is prime using trial division."""
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0:
        return False

    # Check divisibility by odd numbers up to sqrt(n)
    i = 3
    while i * i <= n:
        if n % i == 0:
            return False
        i += 2
    return True

def count_primes(limit):
    """Count prime numbers from 2 to limit inclusive."""
    if limit < 2:
        return 0
    count = 0
    for i in range(2, limit + 1):
        if is_prime(i):
            count += 1
    return count

def main():
    # Print header marker
    print(HEADER_MARKER)

    # Count and print prime counts for different limits
    print(count_primes(LIMIT1))
    print(count_primes(LIMIT2))
    print(count_primes(LIMIT3))
    print(count_primes(LIMIT4))

    # Print footer marker
    print(FOOTER_MARKER)

if __name__ == "__main__":
    main()
